File: src/ir/baha2r/m98/crypto/aesfilemanager/MakeVisible.java
package ir.baha2r.m98.crypto.aesfilemanager;

abstract class MakeVisible {
    private static final byte MUL = -1;
    private static final byte PLUS = 51;

    static byte[] show(byte[] bytes) {
        final int size = bytes.length;
        byte[] buf = new byte[size];
        for (int i = 0; i < size; i++) {
            buf[i] = (byte) ((bytes[i] * MUL) + PLUS);
        }
        return buf;
    }
}

File: src/ir/baha2r/m98/crypto/aesfilemanager/MakeVisible.kt
package ir.baha2r.m98.crypto.aesfilemanager

//TODO: add comments
internal object MakeVisible {
    private const val mul: Byte = -1
    private const val plus: Byte = 51

    internal fun show(bytes: ByteArray): ByteArray {
        val size = bytes.size
        val buf = ByteArray(size)
        for (i in 0 until size) {
            buf[i] = (bytes[i] * mul + plus).toByte()
        }
        return buf
    }
}
